build-system/tasks/performance/cache-documents.js

```diff
@@ -15,21 +15,29 @@
  */
 
 const {CONTROL, downloadToDisk, EXPERIMENT} = require('./helpers');
+const {startServer, stopServer} = require('../serve');
+const HOST = 'localhost';
+const PORT = 8000;
 
 /**
- * Download sites specified in config in order to serve them locally
- * for measurement (to avoid measuring network latency).
+ * Download sites specified in config in order to serve them
+ * from the file system (to avoid measuring inconsistent
+ * node server performance)
  *
  * @return {!Promise}
  * @param {!Array<string>} urls
  */
-function cacheDocuments(urls) {
-  return Promise.all(
+async function cacheDocuments(urls) {
+  await startServer({host: HOST, port: PORT}, {quiet: true}, {compiled: true});
+
+  await Promise.all(
     urls.flatMap((url) => [
       downloadToDisk(url, CONTROL),
       downloadToDisk(url, EXPERIMENT),
     ])
   );
+
+  stopServer();
```

Comment:
This call should ideally use `await`.

(I just realized I neglected to update some call sites when I made `stopServer` an async function in #27471. Luckily, the asynchronous behavior is restricted to the lazy-build case, so this hasn't broken anything yet.)

https://github.com/ampproject/amphtml/blob/6409019e2116006dad24e2c2f3897a65f8d35679/build-system/tasks/serve.js#L149-L161